src/Raven.Client/Documents/Commands/PutDocumentCommand.cs

```diff
@@ -29,6 +30,9 @@ public override HttpRequestMessage CreateRequest(ServerNode node, out string url
                 }),
             };
 
+            if (Etag.HasValue)
+                request.Headers.IfMatch.Add(new EntityTagHeaderValue($"\"{Etag.Value}\""));
+
```

Comment:
I modified it to be:
request.Headers.TryAddWithoutValidation("If-Match", $"\"{etag.Value}\"");
This will avoid no needed class allocation and some validation code but I don't see a way to avoid string concat as we need to pass a string there.
I thought that we do not care about this in the client side as it is going to make a request to the server.